hbase-server/src/test/java/org/apache/hadoop/hbase/master/procedure/TestManageTableErasureCodingPolicy.java

```diff
@@ -0,0 +1,299 @@
+    UTIL.flush();
+  }
+
+  @AfterClass
+  public static void afterClass() throws Exception {
+    UTIL.shutdownMiniCluster();
+    UTIL.shutdownMiniDFSCluster();
+  }
+
+  @Test
+  public void itValidatesPolicyNameForCreate() {
+    runValidatePolicyNameTest(unused -> EC_TABLE_DESC, Admin::createTable);
+  }
+
+  @Test
+  public void itValidatesPolicyNameForAlter() {
+    runValidatePolicyNameTest(admin -> {
+      try {
+        return admin.getDescriptor(NON_EC_TABLE);
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+    }, Admin::modifyTable);
+  }
+
+  @FunctionalInterface
+  interface ThrowingTableDescriptorConsumer {
+    void accept(Admin admin, TableDescriptor desc) throws IOException;
+  }
+
+  private void runValidatePolicyNameTest(Function<Admin, TableDescriptor> descriptorSupplier,
+    ThrowingTableDescriptorConsumer consumer) {
+    HBaseIOException thrown = assertThrows(HBaseIOException.class, () -> {
+      try (Admin admin = UTIL.getAdmin()) {
+        TableDescriptor desc = descriptorSupplier.apply(admin);
+        consumer.accept(admin,
+          TableDescriptorBuilder.newBuilder(desc).setErasureCodingPolicy("foo").build());
+      }
+    });
+    assertThat(thrown.getMessage(),
+      containsString("Cannot set Erasure Coding policy: foo. Policy not found"));
+
+    thrown = assertThrows(HBaseIOException.class, () -> {
+      try (Admin admin = UTIL.getAdmin()) {
+        TableDescriptor desc = descriptorSupplier.apply(admin);
+        consumer.accept(admin,
+          TableDescriptorBuilder.newBuilder(desc).setErasureCodingPolicy("RS-10-4-1024k").build());
+      }
+    });
+    assertThat(thrown.getMessage(), containsString(
+      "Cannot set Erasure Coding policy: RS-10-4-1024k. The policy must be enabled"));
+
+    // RS-6-3-1024k requires at least 6 datanodes, so should fail write test
+    thrown = assertThrows(HBaseIOException.class, () -> {
+      try (Admin admin = UTIL.getAdmin()) {
+        TableDescriptor desc = descriptorSupplier.apply(admin);
+        consumer.accept(admin,
+          TableDescriptorBuilder.newBuilder(desc).setErasureCodingPolicy("RS-6-3-1024k").build());
+      }
+    });
+    assertThat(thrown.getMessage(), containsString("Failed write test for EC policy"));
+  }
+
+  @Test
+  public void testCreateTableErasureCodingSync() throws IOException {
+    try (Admin admin = UTIL.getAdmin(); Table table = UTIL.getConnection().getTable(EC_TABLE)) {
+      admin.createTable(EC_TABLE_DESC);
+      UTIL.loadTable(table, FAMILY);
+      UTIL.flush(EC_TABLE);
+      Path rootDir = CommonFSUtils.getRootDir(UTIL.getConfiguration());
+      DistributedFileSystem dfs = (DistributedFileSystem) FileSystem.get(UTIL.getConfiguration());
+      checkRegionDirAndFilePolicies(dfs, rootDir, EC_TABLE, "XOR-2-1-1024k", "XOR-2-1-1024k");
+    }
+  }
+
+  @Test
+  public void testModifyTableErasureCodingSync() throws IOException, InterruptedException {
```

Comment:
Could you please add another test similar to `testModifyTableErasureCodingSync()` where we start off with a table with EC and then remove it later?

Reply:
ahh. ignore i see this test already covers the path